extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampSuggestionExtractor.java

```diff
@@ -0,0 +1,51 @@
+// Created by Fynn Godau 2019, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.extractor.services.bandcamp.extractors;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.suggestion.SuggestionExtractor;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.schabi.newpipe.extractor.services.bandcamp.linkHandler.BandcampSearchQueryHandlerFactory.CHARSET_UTF_8;
+
+public class BandcampSuggestionExtractor extends SuggestionExtractor {
+
+    private static final String AUTOCOMPLETE_URL = "https://bandcamp.com/api/fuzzysearch/1/autocomplete?q=";
+    public BandcampSuggestionExtractor(StreamingService service) {
+        super(service);
+    }
+
+    @Override
+    public List<String> suggestionList(String query) throws IOException, ExtractionException {
+        Downloader downloader = NewPipe.getDownloader();
+
+        JSONObject fuzzyResults = new JSONObject(
+                downloader.get(AUTOCOMPLETE_URL + URLEncoder.encode(query, CHARSET_UTF_8)).responseBody()
+        );
+
+        JSONArray jsonArray = fuzzyResults.getJSONObject("auto")
+                .getJSONArray("results");
+
+        ArrayList<String> suggestions = new ArrayList<>();
+
+        for (int i = 0; i < jsonArray.length(); i++) {
```

Comment:
Can't you use `for (Object item : jsonArray)`?

Reply:
No, I can't; it would have to be `for (JsonObject item : jsonArray)` (because I then want to get a `String` from this `JsonObject`) and that is not possible because the Json parser doesn't know of what kind its array is.

This is as if I would replace `getObject(i)` with `get(i)`, which returns an `Object` as well, not a `JsonObject`.

Reply:
For YouTube, we do `for (Object item : jsonArray)` everywhere and just downcast i.e. `item` to a `JsonObject`

Reply:
I changed it (9bc7a470abb3189b45d147b6f09ceb54338bf1d5); I do think the way I did is cleaner, but there's no need to argue about it…